plugins/msg-broker/mcollective/lib/openshift/mcollective_application_container_proxy.rb

```diff
@@ -1986,6 +1988,9 @@ def move_gear_pre(gear, state_map)
           reply.append source_container.force_stop(gear)
         end
       end
+
+      log_debug "DEBUG: Unpublishing routing information for gear '#{gear.uuid}'"
+      gear.unpublish_routing_info
```

Comment:
We should probably do this up front, right before line 1962, so that we tell the load balancer to stop routing to the gear *before* we stop the gear.